games/utils/stepicclient.py

```diff
@@ -24,69 +19,31 @@ class StepicError(Exception):
 
 
 class StepicClient(object):
-    def __init__(self, login, password):
-        self.login = login
-        self.password = password
-        self.session = requests.Session()
-        headers = {
-            'Content-Type': 'application/json',
-            'Referer': STEPIC_URL,
-        }
-        self.session.headers.update(headers)
-        self._is_logged_in = False
-
-    def _request(self, method, url, is_json=True, **kwargs):
-        logger.debug("StepicClient request: %s %s | body: %s",
-                     method.upper(), url, kwargs.get('data'))
-        if is_json and 'data' in kwargs:
-            kwargs['data'] = json.dumps(kwargs['data'])
-        if not is_json:
-            headers = kwargs.get('headers', {})
-            headers['Content-Type'] = 'application/x-www-form-urlencoded'
-            kwargs['headers'] = headers
-
-        res = self.session.request(method, url, **kwargs)
-        if not res:
-            logger.info("StepicClient response: %s %s on request: %s %s | "
-                        "body: %s", res.status_code, res.content,
-                        method.upper(), url, kwargs.get('data'))
-        else:
-            logger.debug("StepicClient response: %s %s on request: %s %s",
-                         res.status_code, res.content, method.upper(), url)
-        return res
-
-    def _login(self):
-        self._request('GET', STEPIC_URL)
-        data = {
-            'csrfmiddlewaretoken': self.session.cookies['csrftoken'],
-            'login': self.login,
-            'password': self.password,
-            'remember': 'on',
-        }
-        res = self._request('POST', STEPIC_LOGIN_URL,
-                            data=data, is_json=False, allow_redirects=False)
-        if res.status_code != 302:
-            raise LoginError()
-        self._is_logged_in = True
-
-    def get_attempt(self, id):
-        if not self._is_logged_in:
-            self._login()
-        res = self._request('GET', STEPIC_API_ATTEMPT_URL.format(id=id))
-        if res.status_code == 404:
+    def __init__(self, client_id, client_secret):
+        # Get a token
+        auth = requests.auth.HTTPBasicAuth(client_id, client_secret)
+        response = requests.post('https://stepic.org/oauth2/token/',
+                                 data={'grant_type': 'client_credentials'},
+                                 auth=auth)
+        self.token = json.loads(response.text)['access_token']
+        self.auth_headers = {'Authorization': 'Bearer ' + self.token}
```

Comment:
Sometimes it's handy to use a requests session and set all the required headers and auth settings on it: http://docs.python-requests.org/en/latest/user/advanced/#session-objects
Then you just use your session with configured headers and don't worry about setting them on each request.
There was a similar behavior already, no idea why you have abandoned that.